query_class_config.hh

```diff
@@ -28,14 +28,23 @@ namespace query {
 struct max_result_size {
     uint64_t soft_limit;
     uint64_t hard_limit;
+    uint64_t page_size = 0;
 
     max_result_size() = delete;
     explicit max_result_size(uint64_t max_size) : soft_limit(max_size), hard_limit(max_size) { }
```

Comment:
This constructor should set `page_size` to `max_size` too, as this callers using this constructor actually mean to set the page size.

Reply:
I guess it could even though the effect will be exactly the same as leaving it with the value of 0. If page_size == 0 then `get_page_size` returns hard_limit.

Reply:
Leaving it equal to 0 is more consistent with the other constructor which takes soft_limit and hard_limit.

Reply:
Ok, but maybe make the fields private and add getters, so we can enforce users accessing it through the getter which has this logic.

Reply:
Makes sense. I wanted to make the field private but was worried that deserialiser is assigning the values directly. It turns out it uses a constructor but I'm still not sure how it fetches the value of the field for serialisation.

Reply:
```
build/dev/gen/idl/read_command.dist.impl.hh:153:22: error: 'page_size' is a private member of 'query::max_result_size'
  serialize(buf, obj.page_size);
```

Reply:
IDL supports specifying a getter for the field in case it is private. In this case, instead of the field name you specify the getter itself.

Reply:
If this causes churn, we can do this in a follow-up.

Reply:
`friend class ser::serializer<query::max_result_size>;` seems to do the trick.